bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -614,13 +603,12 @@ func (r *KubeadmConfigReconciler) resolveSecretFileContent(ctx context.Context,
 
 // ClusterToKubeadmConfigs is a handler.ToRequestsFunc to be used to enqeue
 // requests for reconciliation of KubeadmConfigs.
-func (r *KubeadmConfigReconciler) ClusterToKubeadmConfigs(o handler.MapObject) []ctrl.Request {
+func (r *KubeadmConfigReconciler) ClusterToKubeadmConfigs(o client.Object) []ctrl.Request {
 	result := []ctrl.Request{}
 
-	c, ok := o.Object.(*clusterv1.Cluster)
+	c, ok := o.(*clusterv1.Cluster)
 	if !ok {
-		r.Log.Error(errors.Errorf("expected a Cluster but got a %T", o.Object), "failed to get KubeadmConfigs for Cluster")
-		return nil
+		panic(fmt.Sprintf("Expected a Cluster but got a %T", o))
```

Comment:
Nit: if we're just gonna panic we can do `c := o.Object.(*clusterv1.Cluster)` and we'll get 

```
panic: interface conversion: client.Object is foo, not *cluster.Cluster
```

Reply:
I'd rather keep the panic explicit